### PR TITLE
Extend LossOrMetricFn type

### PR DESCRIPTION
**Pull request by gnodar01**
- add support for cosineDistance loss function signature

To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

### CODE REVIEW

**Comment by gnodar01** [comment]
Seems like it's failing on:

> Step #2 - "run-cloudbuild": Step #5 - "bazel-tests": FAIL: //tfjs-layers:bs_chrome_mac_tfjs-layers_webgl2_test (see /builder/home/.cache/bazel/_bazel_root/eab0d61a99b6696edb3d2aff87b585e8/execroot/tfjs/bazel-out/k8-fastbuild/testlogs/tfjs-layers/bs_chrome_mac_tfjs-layers_webgl2_test/test.log)

but not sure how to access the logs.
